src/components/TransitionSwitch/TransitionSwitch.tsx

```diff
@@ -22,23 +41,33 @@ export class TransitionSwitch extends React.Component<TransitionSwitchProps, und
         const transitionProps: any = {
             ...props,
             ...{
-                key: location.pathname.split("/")[1],
+                key: this.routeProps.key,
             }
         } as any;
 
         const currentRouteProps = {
             ...this.routeProps,
-            ...{
-                location,
-            }
+            ...{location},
         };
 
+        this.setDirection(currentRouteProps.key);
+
         return (
-            <TransitionGroup className={this.props.className}>
+            <TransitionGroup className={`${this.props.className}${this.state.direction}`}>
                 <CSSTransition {...transitionProps}>
                     <Route {...currentRouteProps}/>
                 </CSSTransition>
             </TransitionGroup>
         );
     }
+
+    protected setDirection(key: number) {
+        if (this.previousRouteKey === key) {
+            return;
+        }
+
+        this.state.direction = this.previousRouteKey > this.routeProps.key ? " down" : " up";
```

Comment:
May be this.state.direction -> this.state.directionClassName to make clear that this is className?